src/parser/mod.rs

```diff
@@ -3742,24 +3742,23 @@ impl<'a> Parser<'a> {
             });
         }
         self.expect_token(&Token::LParen)?;
-        let in_op = if self.parse_keyword(Keyword::SELECT) || self.parse_keyword(Keyword::WITH) {
-            self.prev_token();
-            Expr::InSubquery {
+        let in_op = match self.try_parse(|p| p.parse_query_body(p.dialect.prec_unknown())) {
```

Comment:
Can we use `self.maybe_parse()` instead of `try_parse()`? that would automatically handle the recursion exceeded case for example

Reply:
Done! Thank you, that is nicer.